lib/charms/data_platform_libs/v0/data_interfaces.py

```diff
@@ -467,13 +502,46 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+def static_secrets_only(f):
+    """Decorator to ensure that certain operations would be only executed on Juju3."""
+
+    def wrapper(self, *args, **kwargs):
+        if self.static_secret_fields:
+            raise IllegalOperationError(
+                "Unsafe usage of statically and dynamically defined secrets, aborting."
+            )
```

Comment:
Shouldn't it be something like the code below?
```suggestion
        if set(self.current_secret_fields) - set(
            self.static_secret_fields
        ):
            raise IllegalOperationError(
                "Unsafe usage of dynamically defined secrets, aborting."
            )
```
It appears to me that this wrapper is currently raising an exception if we have static secrets.

Reply:
Yes. And that's exactly the point.

See [DA076 - Peer Relation Data: No mixing of “dynamic” and “static” approach
](https://docs.google.com/document/d/1folOFb--11xapLbrnmji7ErbLmagRffbYdiEPj_BTF8/edit#heading=h.9e3keicxygz)

Reply:
Bah, but I get your point -- the name of the decorator should be `dynamic_secrets_only` :sweat_smile: 
(NOT the opposite as it is currently: `static_secrets_only`)

Valid point :sweat_smile: